src/ansys/acp/core/_tree_objects/cut_off_geometry.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from __future__ import annotations
+
+from collections.abc import Iterable
+
+from ansys.api.acp.v0 import cut_off_geometry_pb2, cut_off_geometry_pb2_grpc
+
+from .._utils.property_protocols import ReadWriteProperty
+from ._grpc_helpers.property_helper import (
+    grpc_data_property,
+    grpc_data_property_read_only,
+    grpc_link_property,
+    mark_grpc_properties,
+)
+from .base import CreatableTreeObject, IdTreeObject
+from .enums import (
+    CutOffGeometryOrientationType,
+    cut_off_geometry_orientation_type_from_pb,
+    cut_off_geometry_orientation_type_to_pb,
+    status_type_from_pb,
+)
+from .object_registry import register
+from .virtual_geometry import VirtualGeometry
+
+__all__ = ["CutOffGeometry"]
+
+
+@mark_grpc_properties
+@register
+class CutOffGeometry(CreatableTreeObject, IdTreeObject):
+    """Instantiate a cut-off geometry.
+
+    Parameters
+    ----------
+    name :
+        Name of the cut-off geometry.
+    active :
+        Inactive cut-off geometries are not used in the solid model extrusion.
+    cad_geometry :
+        The geometry defining the cut-off.
+    orientation :
+        Determines the cutting orientation of a surface/body geometry. Allows to
+        switch between include and exclude.
+    relative_merge_tolerance :
+        Set the merging tolerance for neighboring nodes relative to the element size.
+    """
+
+    __slots__: Iterable[str] = ()
+
+    _COLLECTION_LABEL = "cut_off_geometries"
+    _OBJECT_INFO_TYPE = cut_off_geometry_pb2.ObjectInfo
+    _CREATE_REQUEST_TYPE = cut_off_geometry_pb2.CreateRequest
+    _SUPPORTED_SINCE = "25.1"
+
+    def __init__(
+        self,
+        *,
+        name: str = "CutOffGeometry",
+        active: bool = True,
+        cad_geometry: VirtualGeometry | None = None,
+        orientation: CutOffGeometryOrientationType = CutOffGeometryOrientationType.UP,
```

Comment:
The suffix _type is used where the parameter if of type enum. This also highlights that specific options are allowed. 
Occurrences of `_type`  where we are not following this rule are `relative_rule_type`, `include_rule_type`.

I would call this parameter orientation_type, and rename `relative_rule_type` and `include_rule_type` to `relative_rule` and `include_rule` since these are booleans.

Reply:
I had a look at the `enum_types.rst`, and how they are used. It seems at least in this direction, there are quite a few exceptions to this convention:

```
cut_off_material_handling: CutOffMaterialType
draping_material_model: DrapingMaterialType
draping: DrapingType
drop_off_material_handling: DropOffMaterialType
extrusion_method: ExtrusionMethodType
format: FeFormat
draping: ImportedPlyDrapingType
linked_object_handling: LinkedObjectHandling
interpolation_algorithm: LookUpTable3DInterpolationAlgorithm
offset_direction: OffsetDirectionType
format: PlyGeometryExportFormat
rosette_selection_method: RosetteSelectionMethod
format: SolidModelExportFormat
format: SolidModelImportFormat
format: SolidModelSkinExportFormat
status: StatusType
unit_system: UnitSystemType
dimension: VirtualGeometryDimension
```

Some of them maybe just need changing, but for others (e.g. `format` or `rosette_selection_method`) I think the current shorter / more descriptive name is better.

Reply:
```
cut_off_material_handling: CutOffMaterialType -> rename enum
draping_material_model: DrapingMaterialType -> rename enum
draping: DrapingType -> rename parameter
drop_off_material_handling: DropOffMaterialType -> rename enum
extrusion_method: ExtrusionMethodType -> rename enum
format: FeFormat -> ok
draping: ImportedPlyDrapingType -> rename parameter
linked_object_handling: LinkedObjectHandling -> ok
interpolation_algorithm: LookUpTable3DInterpolationAlgorithm -> ok
offset_direction: OffsetDirectionType -> offset_direction_type: SolidModelOffsetDirectionType
format: PlyGeometryExportFormat -> ok
rosette_selection_method: RosetteSelectionMethod -> ok
format: SolidModelExportFormat -> ok
format: SolidModelImportFormat -> ok
format: SolidModelSkinExportFormat -> ok
status: StatusType -> rename enum
unit_system: UnitSystemType -> ok
dimension: VirtualGeometryDimension -> ok
```
TODO in a separate PR, along with the ones mentioned above (`include_rule`, `relative_rule`).

Reply:
Converted to issue #658.